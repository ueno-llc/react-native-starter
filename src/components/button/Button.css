.host {
  height: 52px;
  background-color: #E0E3E6;
  border-radius: 4px;
  align-items: center;
  justify-content: center;
  margin-bottom: 20px;
}

.container {
  flex: 1;
  padding: 0 60px;
  align-items: center;
  justify-content: center;
  background-color: transparent;
}

.text {
  text-align: center;
}
