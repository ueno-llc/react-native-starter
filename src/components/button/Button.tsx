import * as React from 'react';
import * as PropTypes from 'prop-types';
import { StyleSheet, View, Text, TouchableOpacity, GestureResponderEvent } from 'react-native';
const styles = require('./Button.css');

interface Props {
  children?: React.ReactNode;
  onPress?: (event: GestureResponderEvent) => void;
  testID?: string;
}

export default class Button extends React.Component<Props> {
  render() {
    const { children, onPress, testID } = this.props;

    return (
      <TouchableOpacity
        accessibilityComponentType="button"
        testID={testID}
        onPress={onPress}
        disabled={!onPress}
        style={styles.host}
      >
        <View style={styles.container}>
          <Text style={styles.text}>{children}</Text>
        </View>
      </TouchableOpacity>
    );
  }
}
