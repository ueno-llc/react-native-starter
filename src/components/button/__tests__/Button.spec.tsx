import 'react-native';
import * as React from 'react';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';
import Button from '../Button';

test('renders correctly', () => {
  const button = <Button />;
  const tree = renderer.create(button).toJSON();

  expect(tree).toMatchSnapshot();
});

test('Button renders children text', () => {
  const text = 'Sample text';
  const button = shallow(<Button>{text}</Button>);

  expect(button.findWhere(c => c.text() === text)).toBeTruthy();
});
