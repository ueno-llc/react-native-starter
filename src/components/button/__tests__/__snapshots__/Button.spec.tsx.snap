// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityComponentType="button"
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E0E3E6",
      "borderRadius": 4,
      "height": 52,
      "justifyContent": "center",
      "marginBottom": 20,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingBottom": 0,
        "paddingLeft": 60,
        "paddingRight": 60,
        "paddingTop": 0,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "textAlign": "center",
        }
      }
    />
  </View>
</View>
`;
